packages/adblocker/src/resources.ts

```diff
@@ -36,129 +65,212 @@ export interface Resource {
-        const name = split[0];
-        const type = split[1];
-        const body = resource.slice(firstNewLine + 1);
-
-        if (name === undefined || type === undefined || body === undefined) {
-          continue;
-        }
+      redirect.content = buffer.getUTF8();
+      redirect.contentType = buffer.getUTF8();
 
-        let resources = typeToResource.get(type);
-        if (resources === undefined) {
-          resources = new Map();
-          typeToResource.set(type, resources);
+      switch (buffer.getUint8() as RedirectResourceEncodingMapping) {
+        case RedirectResourceEncodingMapping.BASE64: {
+          redirect.encoding = 'base64';
+          break;
         }
-        resources.set(name, body);
       }
+
+      redirects.push(redirect);
     }
 
-    // The resource containing javascirpts to be injected
-    const js: Map<string, string> = typeToResource.get('application/javascript') || new Map();
-    for (const [key, value] of js.entries()) {
-      if (key.endsWith('.js')) {
-        js.set(key.slice(0, -3), value);
+    for (let i = 0, l = buffer.getUint32(); i < l; i++) {
+      const scriptlet: ScriptletResource = {
+        names: [],
+        content: '',
+        dependencies: [],
+        executionWorld: 'main',
+        requiresTrust: false,
+      };
+
+      for (let i = 0, l = buffer.getUint32(); i < l; i++) {
+        scriptlet.names.push(buffer.getASCII());
       }
+
+      scriptlet.content = buffer.getUTF8();
+
+      for (let i = 0, l = buffer.getUint32(); i < l; i++) {
+        scriptlet.dependencies.push(buffer.getASCII());
+      }
+
+      switch (buffer.getUint8() as ScriptletResourceExecutionWorldMapping) {
+        case ScriptletResourceExecutionWorldMapping.ISOLATED: {
+          scriptlet.executionWorld = 'isolated';
+          break;
+        }
+      }
+
+      scriptlet.requiresTrust = buffer.getBool();
+
+      scriptlets.push(scriptlet);
     }
 
-    // Create a mapping from resource name to { contentType, data }
-    // used for request redirection.
-    const resourcesByName: Map<string, Resource> = new Map();
-    typeToResource.forEach((resources, contentType) => {
-      resources.forEach((resource: string, name: string) => {
-        resourcesByName.set(name, {
-          contentType,
-          body: resource,
-        });
-      });
-    });
-
-    return new Resources({
-      checksum,
-      js,
-      resources: resourcesByName,
-    });
+    return new this({ checksum, redirects, scriptlets });
   }
 
   public readonly checksum: string;
-  public readonly js: Map<string, string>;
-  public readonly resources: Map<string, Resource>;
 
-  constructor({ checksum = '', js = new Map(), resources = new Map() }: Partial<Resources> = {}) {
+  public readonly redirects: Map<string, RedirectResource>;
+  public readonly scriptlets: Map<string, ScriptletResource>;
+
+  private readonly redirectDetails: Map<string, RedirectDetail>;
```

Comment:
We may only use `redirectDetails` in memory as `redirects` is only here to provide a baseline information to generate redirect detail.